src/main/java/no/unit/nva/doi/CrossRefClient.java

```diff
@@ -39,15 +39,24 @@ public CrossRefClient(HttpClient httpClient) {
         this.httpClient = httpClient;
     }
 
-    public Optional<String> fetchDataForDoi(String doiIdentifier) throws URISyntaxException {
+    /**
+     * The method returns the object containing the metadata (title,author etc) of the publication with the specific
+     * DOI, and the source where the metadata were acquired.
+     *
+     * @param doiIdentifier a valid doi identifier or URL.
+     * @return FetchResult contains the json object and the location from where it was fetched (Datacite, Crossref).
```

Comment:
"JSON", also do we need to enumerate the sources, if we actually return a "location", i.e. a URI?